src/Symfony/Component/DependencyInjection/Compiler/AbstractRecursivePass.php

```diff
@@ -36,6 +36,11 @@ abstract class AbstractRecursivePass implements CompilerPassInterface
     private ExpressionLanguage $expressionLanguage;
     private bool $inExpression = false;
 
+    public function __construct(
+        private bool $skipScalars = true,
```

Comment:
this cannot use CPP because child classes are not required to call this constructor (and the property would remain undefined as is)

in order to preserve BC, the property should default to false, and all passes that don't care about scalars should explicitly call this with `true`

Reply:
could be a protected property that is redefine in child classes BTW, to save the boilerplate of calling the parent constructor